### DISCUSSION TITLE
Brainstorming

### DISCUSSION THREAD

**Discussion by storrisi** [tomorrowdevs-projects/real-world-projects#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

We would like to receive ideas about projects, improvements of this repo and so on.

Can't wait to read about you!!

**Comment by Cadienvan** [comment]
I had an idea popping into my mind and, as promised @storrisi , here it is!

A nice project for people approaching development could be a **form builder.**

On the back-end, you should be able to create a new "Form" entity with customizable fields into it (Let's say a text, a textarea, a select, a multiple choice checkbox, to keep it simple) and apply validators to them (required, regex, minLength, maxLength, etc..).
On the front-end side, you should be able to receive such contract (Let's say a structured JSON containing all the information about your form) and expose it visually.

## Both sides Next Steps
- Provide some sort of dependency mechanism. "This question is visible only if that other question's answer is X".

## Back-end Next Steps
- The first nice next step could be to add a "geolocation" field which could leverage Google Maps APIs or some other source.
- Another cool step is to add custom eval validators. I know eval is mostly an anti-pattern but this could be a nice use case just to learn more about the API. So, let's say, you could write a custom function handling your input and validating it.

## Front-end Next Steps
- A cool step could be the ability to share and embed the component exposing the form everywhere, in any compatible framework or library (Let's say the React component in any React library, or Vue or Angular just to give some examples).



I hope you'll enjoy it!

**Reply by tomorrowdevsit** [reply]
Just to summarize, the idea is to create a library that generates a form starting from a backend configuration, composed by visual informations and validations?

**Reply by Cadienvan** [reply]
Yes, as per the front-end side, it is!

**Reply by tomorrowdevsit** [reply]
Very cool idea @Cadienvan , you can try to brief all the requirements, and even some database configuration if needed, in a new PR :)
